docs/architecture/adr-069-unordered-account.md

```diff
@@ -0,0 +1,177 @@
+# ADR 070: Un-Ordered Transaction Inclusion
+
+## Changelog
+
+* Dec 4, 2023: Initial Draft
+
+## Status
+
+Proposed
+
+## Abstract
+
+We propose a way to do replay-attack protection without enforcing the order of transactions, it don't use nonce at all. In this way we can support un-ordered transaction inclusion.
+
+## Context
+
+As of today, the nonce value (account sequence number) prevents replay-attack and ensures the transactions from the same sender are included into blocks and executed in sequential order. However it makes it tricky to send many transactions from the same sender concurrently in a reliable way. IBC relayer and crypto exchanges are typical examples of such use cases.
+
+## Decision
+
+We propose to add a boolean field `unordered` to transaction body to mark "un-ordered" transactions.
```

Comment:
Maybe a silly Q, but if we're repurposing the `timeout_height` field, do we need a new boolean field? Can we determine if a tx wants to be unordered or not by simply evaluating if `timeout_height` is non-zero?

Reply:
We need to restrict the range of timeout_height of unordered tx, but not for ordered ones.

Reply:
I see. So you're allowing `timeout_height` to be used for it's original purpose and for unordered txs. Makes sense.